src/frontend/src/lib/components/ui/ThreeBackground.svelte

```diff
@@ -0,0 +1,170 @@
+  return 130.0 * dot(m, g);
+}`;
+
+	const vertexShader = `varying vec2 vUv;
+void main(){
+  vUv = uv;
+  gl_Position = vec4(position, 1.);
+}`;
+
+	const fragmentShader = `
+  varying vec2 vUv;
+  uniform float time;
+  uniform float amplitude;
+  uniform float frequency;
+  uniform float speed;
+  uniform vec3 background;
+  uniform vec3 colors[4];
+  uniform float initialNoisePositions[7];
+  uniform float speeds[4];
+  uniform float waveHeight;
+
+  ${simplex2Dshader}
+
+  void main() {
+    float y = vUv.y;
+    float x = vUv.x;
+
+    vec4 finalColor = vec4(background, 1.0);
+
+    for (int i = 0; i < 4; i++) {
+      float noise = snoise(vec2(x * frequency, time * speeds[i] + initialNoisePositions[i]));
+      float wave = waveHeight * (sin(noise * 3.14) + 1.0) / 2.0;
+      float distance = abs(y - wave + .6); // top position
+      float alpha = 1.0 - distance * 2.2;
+      finalColor = mix(finalColor, vec4(colors[i], 1.0), alpha);
+      finalColor.a = 0.2;
+    }
+
+    gl_FragColor = finalColor;
+  }
+`;
+
+	let renderer: WebGLRenderer;
+	let scene: Scene;
+	let camera: PerspectiveCamera;
+	let material: ShaderMaterial;
+	let mesh: Mesh;
+
+	onMount(() => {
+		init();
+		return () => {
+			// Cleanup
+			renderer.dispose();
+			scene.remove(mesh);
+			material.dispose();
+			mesh.geometry.dispose();
+		};
+	});
+
+	const init = () => {
+		scene = new Scene();
+		camera = new PerspectiveCamera(75, container.clientWidth / container.clientHeight, 0.1, 1000);
+		renderer = new WebGLRenderer();
+		renderer.setSize(container.clientWidth, container.clientHeight);
+		container.appendChild(renderer.domElement);
+
+		material = new ShaderMaterial({
+			uniforms: {
+				time: { value: 0.0 },
+				frequency: { value: window.innerWidth < 768 ? 0.5 : 1.0 },
+				background: { value: new Color(colors[2]) },
+				colors: { value: colors.map((c) => new Color(c)) },
+				initialNoisePositions: { value: colors.map(() => Math.random() * 1000) },
+				speeds: { value: colors.map(() => -1 + Math.random() * 2) },
+				waveHeight: { value: 0.5 }
+			},
+			vertexShader,
+			fragmentShader
+		});
+
+		const geometry = new PlaneGeometry(2, 2);
+		mesh = new Mesh(geometry, material);
+		scene.add(mesh);
+
+		camera.position.z = 5;
+
+		animate();
+	};
+
+	const animate = () => {
+		requestAnimationFrame(animate);
+		material.uniforms.time.value += 0.001;
+		renderer.render(scene, camera);
+	};
+
+	const handleResize = () => {
+		if (isNullish(container)) {
+			return;
+		}
+		renderer.setSize(container.clientWidth, container.clientHeight);
+		camera.aspect = container.clientWidth / container.clientHeight;
+		camera.updateProjectionMatrix();
+	};
+
+	onMount(() => {
+		window.addEventListener('resize', handleResize);
+		return () => {
+			window.removeEventListener('resize', handleResize);
+		};
+	});
+</script>
+
+<div bind:this={container} class="three-background"></div>
+
+<style>
+	.three-background {
+		position: absolute;
```

Comment:
we cannot solve the position and size with Tailwind? Like `inset-full` or something?

Reply:
indeed! same comment as above